### PR TITLE
feat: update the status colors of instances

### PR DESCRIPTION
**Pull request by Chisomchima**
implements [DEVOPS-304](https://dhis2.atlassian.net/browse/DEVOPS-304)
---

### Description
updated the instance status color to suit specifications for every deployment.

### CODE REVIEW

File: src/views/instances/details/status-label.tsx [hunk 1]
```diff
@@ -6,14 +6,15 @@ import styles from './status-label.module.css'
 type Status = 'NotDeployed' | 'Pending' | 'Booting' | 'Booting (%d)' | 'Running' | 'Error'
 
 const getTagProps = (status: Status) => {
-    if (status?.startsWith('Booting')) {
-        return { neutral: true } // blue
-    } else if (status === 'Error' || status === 'NotDeployed') {
-        return { negative: true } // red
+    if (!status) return
+    if (status.includes('Booting') || status === 'Pending') {
```
Comment (tonsV2) [inline src/views/instances/details/status-label.tsx:10]: Why not use `startsWith`? It's more confined

**Review by tonsV2 — CHANGES_REQUESTED** [review]
Please run `yarn format` before you push :pray:

**Review by Mohammer5 — CHANGES_REQUESTED** [review]
When I open the UI with one instance (`feat-public-deployments` as `ENVIRONMENT` in the `.env` file), the instance has two tags (`dhis2-db` and `dhis2-core`).

* Requested change: The url for both statuses is the same, so we could load the status once, per instance, instead of per tag
* Issue when reviewing: The actual request response with a 409 (twice, as there are two tags) with the following message: `multiple pods found using the selector: "im-default=true,im-id=2"`

Not sure why I'm getting an error? (cc @tonsV2)

**Comment by tonsV2** [comment]
> When I open the UI with one instance (`feat-public-deployments` as `ENVIRONMENT` in the `.env` file), the instance has two tags (`dhis2-db` and `dhis2-core`).
> 

There are two tags because there are two instances running within the cluster. Server side an instance is actually called a deployment and consist of multiple instances. Those instances could be dhis2-core as you're used to but also a Postgresql instance, a Redis instance or a MinIO (S3) instance.

> * Requested change: The url for both statuses is the same, so we could load the status once, per instance, instead of per tag

In this case they're the same and hopefully they usually both end up in the same state (running). However it's unlikely before they get there they go through multiple states and often not at the same time. We have talked about showing only an aggregated state one the list view but we haven't implemented it yet.

> * Issue when reviewing: The actual request response with a 409 (twice, as there are two tags) with the following message: `multiple pods found using the selector: "im-default=true,im-id=2"`
> 
> Not sure why I'm getting an error? (cc @tonsV2)

This can happen due to several things but it's most likely because an instance was evicted from one node only to be respawned on a different node.
We haven't updated the "get status" code to deal with this yet.

For the ensuring this is dealt with I've created this [Jira](https://dhis2.atlassian.net/browse/DEVOPS-414) task.

**Comment by Chisomchima** [comment]
> * Requested change: The url for both statuses is the same, so we could load the status once, per instance, instead of per tag
> * Issue when reviewing: The actual request response with a 409 (twice, as there are two tags) with the following message: `multiple pods found using the selector: "im-default=true,im-id=2"`

Each instance contains different tags with different  tag Ids... hence the need to make seperate calls. also a tag represents an instance in a deployment.

File: src/views/instances/list/instances-list.tsx [hunk 2]
```diff
@@ -10,26 +10,25 @@ import {
...
 } from '@dhis2/ui'
 import type { FC } from 'react'
+import { MomentExpiresFromNow } from '../../../components'
 import Moment from 'react-moment'
 import { useNavigate } from 'react-router-dom'
 import { useAuthAxios } from '../../../hooks'
 import { GroupsWithDeployments } from '../../../types'
-import { OpenButton } from './open-button'
 import styles from './instances-list.module.css'
+import { Heading } from '../../../components'
```
[the remark this answers is not in the document]
Comment (Chisomchima) [reply src/views/instances/list/instances-list.tsx]: Nice observation!

File: src/views/instances/list/instances-list.tsx [hunk 3]
```diff
@@ -10,26 +10,25 @@ import {
...
+import InstanceTag from './instance-tag'
+import { OpenButton } from './open-button'
 import { DeleteButton } from './delete-menu-button'
-import { Heading, MomentExpiresFromNow } from '../../../components'
-
```
[the remark this answers is not in the document]
Comment (Chisomchima) [reply src/views/instances/list/instances-list.tsx:26]: done this

**Comment by Chisomchima** [comment]
> Please run `yarn format` before you push 🙏

"After running 'yarn format' and 'yarn lint', I noticed significant changes to files I hadn't modified, suggesting that these commands hadn't been executed in previous commits and are now having a broader impact."

File: src/utils/tag-utils.ts
Comment (tonsV2) [inline src/utils/tag-utils.ts:1]: How about just naming this file "tag.ts". The utils part is redundant :slightly_smiling_face:

**Review by tonsV2 — APPROVED** [review]

File: src/utils/tag-utils.ts
Comment (Chisomchima) [reply src/utils/tag-utils.ts:1]: done

**Review by Mohammer5 — APPROVED** [review]

**Pull request merged**